clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/StoreRepushCommand.java

```diff
@@ -88,14 +109,106 @@ public List<String> getShellCmd() {
+        String latestTimestamp = regionPushDetails.get(destFabric).getPushStartTimestamp();
+        LocalDateTime latestPushStartTime = LocalDateTime.parse(latestTimestamp, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+
+        if (latestPushStartTime.isAfter(timestamp)) {
+          return Pair.of(false, "input timestamp earlier than latest push");
+        }
+
+        boolean isBatch = false;
+        StoreResponse storeResponse = parentCtrlCli.getStore(s);
+        if (storeResponse.getStore().getHybridStoreConfig() == null) {
+          isBatch = true;
+        }
+
+        MultiStoreStatusResponse response = parentCtrlCli.getFutureVersions(clusterName, s);
+        // No future version status for target region.
+        if (!response.getStoreStatusMap().containsKey(destFabric)) {
+          return Pair.of(true, isBatch ? "BATCH" : StringUtils.EMPTY);
+        }
+
+        int futureVersion = Integer.parseInt(response.getStoreStatusMap().get(destFabric));
+        // No ongoing offline pushes detected for target region.
+        if (futureVersion == Store.NON_EXISTING_VERSION) {
+          return Pair.of(true, isBatch ? "BATCH" : StringUtils.EMPTY);
+        }
+        // Find ongoing pushes for this store, skip.
+        return Pair.of(false, String.format("find ongoing push, version: %d", futureVersion));
+      }
+    } catch (Exception e) {
+      return Pair.of(false, "Exception Thrown: " + e.toString());
+    }
   }
 
   @Override
   public void execute() {
+    Pair<Boolean, String> repushViability = getRepushViability();
+    StoreRepushCommand.Params repushParams = getParams();
+    ControllerClient cli = repushParams.getPCtrlCliWithoutCluster();
+    if (repushViability.getLeft() == false) {
+      completeCoreWorkWithMessage("failure: " + repushViability.getRight());
+      return;
+    }
+    if (repushViability.getRight() == "BATCH") {
+      try {
+        String clusterName = cli.discoverCluster(repushParams.getStore()).getCluster();
+        try (ControllerClient parentCtrlCli =
+            buildControllerClient(clusterName, repushParams.getUrl(), repushParams.getSSLFactory())) {
+          ControllerResponse prepareResponse = parentCtrlCli.prepareDataRecovery(
+              repushParams.getSourceFabric(),
+              repushParams.getDestFabric(),
+              repushParams.getStore(),
+              -1,
+              Optional.empty());
+          if (prepareResponse.isError()) {
+            completeCoreWorkWithMessage("failure: " + prepareResponse.getError());
+          }
+          ControllerResponse dataRecoveryResponse = parentCtrlCli.dataRecovery(
+              repushParams.getSourceFabric(),
+              repushParams.getDestFabric(),
+              repushParams.getStore(),
+              -1,
+              false,
+              true,
+              Optional.empty());
+          if (dataRecoveryResponse.isError()) {
+            completeCoreWorkWithMessage("failure: " + dataRecoveryResponse.getError());
```

Comment:
For failure cases, let use something similar to `completeCoreWorkWithError` for reporting.

```
private void completeCoreWorkWithError(String error) {
    result.setError(error);
    result.setCoreWorkDone(true);
  }
```